File: easy/3550-smallest-index-with-digit-sum-equal-to-index/solution.java
/**
 * 3550. Smallest Index With Digit Sum Equal to Index
 * Time: O(n * d)
 * Space: O(1)
 */
class Solution {
    public int smallestIndex(int[] nums) {
        for (int i = 0; i < nums.length; i++) {
            int v = nums[i];
            int digitSum = 0;
            while (v > 0) {
                digitSum += v % 10;
                v /= 10;
            }
            if (digitSum == i) {
                return i;
            }
        }
        return -1;
    }
}

File: easy/3550-smallest-index-with-digit-sum-equal-to-index/solution.py
from typing import List


class Solution:
    def smallestIndex(self, nums: List[int]) -> int:
        """
        3550. Smallest Index With Digit Sum Equal to Index
        Time: O(n * d)
        Space: O(1)
        """
        for i, v in enumerate(nums):
            digit_sum = 0
            while v > 0:
                digit_sum += v % 10
                v //= 10
            if digit_sum == i:
                return i
        return -1
